ui/src/App.test.js

```diff
@@ -1,8 +1,110 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import { fireEvent, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import App from "./App";
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+describe("App", () => {
+  beforeAll(() => {
+    jest.useFakeTimers("modern");
+    const date = new Date("2022-02-15T04:00");
+    jest.setSystemTime(date);
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it("renders correctly", () => {
+    const app = render(<App />);
+    expect(app.asFragment()).toMatchSnapshot();
+  });
+
+  it("shows calander", async () => {
+    render(<App />);
+    expect(await screen.findByText("Month")).toBeVisible();
+  });
+
+  describe("events", () => {
+    it("defaults to today's date and a one hour time window", () => {
+      render(<App />);
+      expect(screen.getByLabelText("Start Date")).toHaveValue("2022-02-15");
+      expect(screen.getByLabelText("End Date")).toHaveValue("2022-02-15");
+      expect(screen.getByLabelText("Start Time")).toHaveValue("04:00");
+      expect(screen.getByLabelText("End Time")).toHaveValue("05:00");
+    });
+
+    it("displays event in ui when all inputs are provided valid values", async () => {
+      render(<App />);
+      userEvent.click(screen.getByLabelText("Title"));
+      userEvent.type(
+        screen.getByLabelText("Title"),
+        "Berta goes to the baseball game!"
+      );
+
+      userEvent.click(screen.getByRole("button", { name: "Create Event" }));
+      expect(await screen.findByLabelText("Title")).toHaveAttribute(
+        "value",
+        ""
+      );
+
+      expect(
+        screen.getByText("Berta goes to the baseball game!")
+      ).toBeVisible();
+    });
+
+    it("resets inputs correclty to default values when submitted", () => {
+      render(<App />);
+      userEvent.type(screen.getByLabelText("Start Date"), "2022-03-16");
+      userEvent.type(screen.getByLabelText("Start Time"), "08:10");
+      userEvent.type(screen.getByLabelText("End Date"), "2022-03-18");
+      userEvent.type(screen.getByLabelText("End Time"), "10:10");
+      userEvent.click(screen.getByRole("button", { name: "Create Event" }));
+
+      expect(screen.getByLabelText("Start Date")).toHaveValue("2022-02-15");
+      expect(screen.getByLabelText("End Date")).toHaveValue("2022-02-15");
+      expect(screen.getByLabelText("Start Time")).toHaveValue("04:00");
+      expect(screen.getByLabelText("End Time")).toHaveValue("05:00");
+    });
+
+    it("errors when end date is before start date", () => {
+      render(<App />);
+      fireEvent.change(screen.getByLabelText("Start Date"), {
```

Comment:
Since you used `userEvent` actions in the test above, you could re-write these to use userEvent instead of fireEvent as well. Or, change the test above to use fireEvent. Either way, let's just be consistent

Reply:
userEvent's wording feels nice to me and it's visually shorter than fireEvent so I updated them all to userEvent